modules/indexing/src/test/java/org/apache/ignite/internal/processors/cache/index/BasicIndexTest.java

```diff
@@ -1984,6 +1988,100 @@ public void testCreateSystemIndexWithSpecifiedInlineSizeByDdl() throws Exception
         assertTrue(lsnr.check());
     }
 
+    /** */
+    @Test
+    public void testCreateIndexFailsWhenInlineSizeInDdlExceedsMax() throws Exception {
+        inlineSize = -1;
+
+        startGrid();
+
+        sql("CREATE TABLE TEST (ID VARCHAR, ID_AFF INT, VAL INT, PRIMARY KEY (ID, ID_AFF))");
+
+        // We need a value, that exceeds calculated calculated max allowed inline size for sure.
+        int idxInlineSize = PageIO.MAX_PAYLOAD_SIZE;
+        assertThrows(
+                log,
+                () -> sql("CREATE INDEX FAILING_IDX ON TEST (VAL) inline_size " + idxInlineSize),
+                IgniteSQLException.class,
+                "Inline size is too big [cacheName=TEST" +
+                        ", tableName=SQL_PUBLIC_TEST" +
+                        ", idxName=FAILING_IDX" +
+                        ", configuredInlineSize=" + idxInlineSize +
+                        ", maxAllowedInlineSize="
+        );
```

Comment:
Also, 
- let's check the cluster is operable after the error.
- let's check system view shows correct inline size value on server and client (and create a ticket to fix this)
- let's check client and server works fine, when page size on server side is larger than default 4k. E.g. 8k.